pkg/endpoint/endpoint.go

```diff
@@ -1873,11 +1863,6 @@ func (e *Endpoint) SetStateLocked(toState, reason string) bool {
 	// Validate the state transition.
 	fromState := e.state
 	switch fromState { // From state
-	case StateCreating:
```

Comment:
We should not be removing `StateCreating` for backward compatibility. The earlier api still allows `StateCreating` 
CC @rlenglet I asked Romain the same question in the other PR #3863 I was fixing, and he asked me to not remove this state to maintain back ward api compatibility.

Reply:
Let's isolate the change of removing the Creating state and dedicate a separate PR for that. I don't understand the implications yet.

Reply:
Sure, will do.